# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../apr/apr-config.patch.cross
# Copyright (C) 2009 - 2010 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- apr-1.3.2/apr-config.in.vanilla	2009-11-20 18:45:07.000000000 +0100
+++ apr-1.3.2/apr-config.in	2009-11-20 18:58:55.000000000 +0100
@@ -91,7 +91,7 @@
 fi
 
 if test "$location" = "installed"; then
-    LA_FILE="$libdir/lib${APR_LIBNAME}.la"
+    LA_FILE=""
 else
     LA_FILE="$APR_BUILD_DIR/lib${APR_LIBNAME}.la"
 fi
@@ -152,7 +152,7 @@
     ;;
     --includes)
     if test "$location" = "installed"; then
-        flags="$flags -I$includedir $EXTRA_INCLUDES"
+        flags="$flags $EXTRA_INCLUDES"
     elif test "$location" = "source"; then
         flags="$flags -I$APR_SOURCE_DIR/include $EXTRA_INCLUDES"
     else
@@ -166,7 +166,7 @@
     ;;
     --installbuilddir)
     if test "$location" = "installed"; then
-        echo "${installbuilddir}"
+        echo "`dirname $0`/../../${installbuilddir}"
     elif test "$location" = "source"; then
         echo "$APR_SOURCE_DIR/build"
     else
@@ -182,7 +182,7 @@
     --link-ld)
     if test "$location" = "installed"; then
         ### avoid using -L if libdir is a "standard" location like /usr/lib
-        flags="$flags -L$libdir -l${APR_LIBNAME}"
+        flags="$flags -l${APR_LIBNAME}"
     else
         ### this surely can't work since the library is in .libs?
         flags="$flags -L$APR_BUILD_DIR -l${APR_LIBNAME}"
@@ -199,7 +199,7 @@
         ### avoid using -L if libdir is a "standard" location like /usr/lib
         # Since the user is specifying they are linking with libtool, we
         # *know* that -R will be recognized by libtool.
-        flags="$flags -L$libdir -R$libdir -l${APR_LIBNAME}"
+        flags="$flags -l${APR_LIBNAME}"
     else
         flags="$flags $LA_FILE"
     fi
@@ -223,7 +223,7 @@
     ;;
     --apr-libtool)
     if test "$location" = "installed"; then
-        echo "${installbuilddir}/libtool"
+        echo "`dirname $0`/../../${installbuilddir}/libtool"
     else
         echo "$APR_BUILD_DIR/libtool"
     fi
